pkg/rbd/controllerserver.go

```diff
@@ -177,6 +177,14 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		return nil, err
 	}
 
+	if rbdVol.Encrypted {
+		// format the image with cryptsetup
+		if err = encryptDevice(ctx, rbdVol, req.GetSecrets(), cr); err != nil {
```

Comment:
I think it would be a good idea to persistently record that the image is encrypted. This will be needed for properly handling (or disabling) snapshots and cloning operations.

Reply:
what do I need to add to persist this information? I thought the volOptions structure is saved in configmaps

Reply:
> what do I need to add to persist this information? 

You can store it in the RBD image-meta fields via `rbd image-meta set <key> <value>`, where perhaps `key` is something like `.rbd.csi.ceph.com/encrypted`. Then snapshot/clone operations can check for this and disallow for now.

> I thought the volOptions structure is saved in configmaps

Not to my knowledge -- that might have been the legacy behaviour.

Reply:
would another field in CSIJournal be the right place for this? https://github.com/ceph/ceph-csi/blob/8e117db80d197c46e4205352caada92b8e490085/pkg/util/voljournal.go#L97

I don't see `image-meta` to be used anywhere else, but if journal is not a right place for such kind of information I can save it in image-meta

Reply:
The `CSIJournal` is a directory used for mapping RBD <-> CO names so I don't think it would be a good place to store other attributes.

Reply:
ok, in that case I'll use image-meta for this as you've suggested